docs/articles/documentation/extending-testcafe/creating-custom-reporter/reporterpluginhost.md

```diff
@@ -0,0 +1,280 @@
+
+    if (!this.afterErrList)
+        this.newline();
+
+    this.newline()
+        .write(footer)
+        .newline();
+}
+```
+
+### createErrorDecorator
+
+Allows implementing a custom error decorator that will change default error colors.
+
+The methods is added to `src/index.js` if you enable a custom error decorator when [generating the reporter](/testcafe/documentation/extending-testcafe/creating-custom-reporter/#generating-reporter-project).
+
+**Example**
+
+```js
+createErrorDecorator () {
+    return {
+        'span category':       () => '',
+        'span step-name':      str => `"${str}"`,
+        'span user-agent':     str => this.chalk.gray(str),
+        'div screenshot-info': str => str,
+        'a screenshot-path':   str => this.chalk.underline(str),
+        'code':                str => this.chalk.yellow(str),
+        'code step-source':    str => this.chalk.magenta(this.indentString(str, 4)),
+        'span code-line':      str => `${str}\n`,
+        'span last-code-line': str => str,
+        'code api':            str => this.chalk.yellow(str),
+        'strong':              str => this.chalk.cyan(str),
+        'a':                   str => this.chalk.yellow(`"${str}"`)
+    };
+}
+```
+
+## Helper Methods
+
+To format a report output, you can use the following methods.
+
+### String Helpers
+
+#### indentString
+
+Indents each line in a string by a space.
+
+```js
+indentString (str, indentVal)
+```
+
+**str** - the string you want to indent.
+
+**indentVal** - the number of spaces to indent a new line.
+
+#### wordWrap
+
+Allows long strings to be able to break and wrap onto the next line.
+If the string length is longer than maximum allowed, the string is broken.
+
+```js
+wordWrap (str, indentVal, width)
+```
+
+**str** - the string you want to break.
+
+**indentVal** - the number of spaces to indent a new line.
+
+**width** - the maximum number of characters each line can contain.
+
+#### escapeHtml
+
+Escapes string for use in HTML.
+
+```js
+escapeHtml (str)
+```
+
+**str** - the string you want to escape.
+
+#### formatError
+
+Returns the formatted error message string for the specified error.
+
+```js
+formatError (err, prefix = '')
+```
+
+**err** - the error you need to format.
+
+**prefix = ''** - the string that is prepended to the error. By default, it is empty.
+
+**Example**
+
+```js
+this.formatError(err, '')
+```
+
+### Writing Helpers
+
+#### newline
+
+Adds a new line to the report.
+
+```js
+newline ()
+```
+
+#### write
+
+Writes the specified text in the report.
+
+```js
+write (text)
+```
+
+**text** - the text to output in the report.
+
+#### useWordWrap
+
+Enables word wrapping within the report.
+
+```js
+useWordWrap (use)
+```
+
+**use** - Boolean. *true* to enable word wrapping.
+
+#### setIndent
+
+Enables indenting by specifying the number of indenting spaces.
+
+```js
+setIndent (val)
+```
+
+**val** - Integer. Specifies the number of spaces to indent a new line.
```

Comment:
All this methods needs examples and it requires metnion that they are chainable. Also, API description is not consistent with the format used here: https://github.com/DevExpress/testcafe/blob/master/docs/articles/documentation/using-testcafe/programming-interface/Runner.md]